contrib/example-backend/controllers/servicebindingrequest/serviceexport_reconcile.go

```diff
@@ -0,0 +1,142 @@
+	logger := klog.FromContext(ctx)
+	var errs []error
+
+	if req.Status.Export == "" {
+		// create new export
+		export := &kubebindv1alpha1.APIServiceExport{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: req.Name,
+			},
+			Spec: kubebindv1alpha1.APIServiceExportSpec{
+				Scope: kubebindv1alpha1.ClusterScope, // TODO: implement namespace scope
+			},
+		}
+		for _, res := range req.Spec.Resources {
+			export.Spec.Resources = append(export.Spec.Resources, kubebindv1alpha1.APIServiceExportGroupResource{
+				GroupResource: res.GroupResource,
+			})
+		}
+		logger.V(2).Info("Creating service export %s/%s", export.Namespace, export.Name)
+		var err error
+		if _, err = r.createServiceExport(ctx, export); err != nil && !apierrors.IsAlreadyExists(err) {
+			export.GenerateName = req.Name + "-"
+			export.Name = ""
+
+			logger.V(2).Info("Creation of service export %s/%s failed. Trying with a generated name", export.Namespace, export.Name)
+			if _, err = r.createServiceExport(ctx, export); err != nil {
+				errs = append(errs, err)
+				return utilerrors.NewAggregate(errs)
+			}
+		}
+		if err != nil {
+			errs = append(errs, err)
+			return utilerrors.NewAggregate(errs)
+		}
+
+		req.Status.Export = export.Name // waiting for export to be ready for phase update
+		return nil
+	}
+
+	export, err := r.getServiceExport(req.Namespace, req.Status.Export)
+	if err != nil && !apierrors.IsNotFound(err) {
+		errs = append(errs, err)
+		return utilerrors.NewAggregate(errs)
+	} else if apierrors.IsNotFound(err) {
+		conditions.MarkFalse(
+			export,
+			kubebindv1alpha1.APIServiceBindingRequestConditionExportsReady,
+			"ServiceExportNotFound",
+			conditionsapi.ConditionSeverityError,
+			"APIServiceExport %s in the service provider cluster not found",
+			req.Status.Export,
+		)
+		return nil
+	}
+
+	if c := conditions.Get(export, conditionsapi.ReadyCondition); c == nil {
+		conditions.MarkFalse(
+			export,
+			kubebindv1alpha1.APIServiceBindingRequestConditionExportsReady,
+			"Unknown",
+			conditionsapi.ConditionSeverityInfo,
+			"APIServiceExport %s in the service provider cluster has no Ready condition",
+			req.Status.Export,
+		)
+		return nil
+	} else if c.Status != corev1.ConditionTrue {
+		conditions.MarkFalse(
+			export,
+			kubebindv1alpha1.APIServiceBindingRequestConditionExportsReady,
+			c.Reason,
+			c.Severity,
+			c.Message,
+			req.Status.Export,
+		)
+		return nil
+	}
+	conditions.MarkTrue(export, kubebindv1alpha1.APIServiceBindingRequestConditionExportsReady)
+
+	if req.Status.Phase == kubebindv1alpha1.APIServiceBindingRequestPhasePending {
+		if time.Since(req.CreationTimestamp.Time) > time.Minute {
+			req.Status.Phase = kubebindv1alpha1.APIServiceBindingRequestPhaseFailed
+		}
+		req.Status.Phase = kubebindv1alpha1.APIServiceBindingRequestPhaseSucceeded
+	} else if time.Since(req.CreationTimestamp.Time) > 10*time.Minute {
+		return r.deleteServiceBindingRequest(ctx, req.Namespace, req.Status.Export)
```

Comment:
worth an event or log?

Reply:
yes, all actions should be logged.

We don't have events yet. Worth an issue.

Reply:
will add 👍